org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/markers/MarkerUtils.java

```diff
@@ -71,4 +72,47 @@ public static void updateAllSonarMarkerSeverity() throws CoreException {
     }
   }
 
+  public static FlatTextRange toFlatTextRange(final IDocument document, TextRange textRange) {
+    if (textRange == null || textRange.getStartLine() == null) {
+      return null;
+    }
+    if (textRange.getStartLineOffset() == null) {
+      return toFlatTextRange(document, textRange.getStartLine());
+    }
+    return toFlatTextRange(document, textRange.getStartLine(), textRange.getStartLineOffset(), textRange.getEndLine(), textRange.getEndLineOffset());
+  }
+
+  public static FlatTextRange toFlatTextRange(final IDocument document, int startLine) {
+    int startLineStartOffset;
+    int length;
+    String lineDelimiter;
+    try {
+      startLineStartOffset = document.getLineOffset(startLine - 1);
+      length = document.getLineLength(startLine - 1);
+      lineDelimiter = document.getLineDelimiter(startLine - 1);
+    } catch (BadLocationException e) {
+      return null;
+    }
+
+    int lineDelimiterLength = lineDelimiter != null ? lineDelimiter.length() : 0;
+
+    int start = startLineStartOffset;
+    int end = startLineStartOffset + length - lineDelimiterLength;
+    return new FlatTextRange(start, end);
+  }
+
+  private static FlatTextRange toFlatTextRange(final IDocument document, int startLine, int startLineOffset, int endLine, int endLineOffset) {
+    int startLineStartOffset;
+    int endLineStartOffset;
+    try {
+      startLineStartOffset = document.getLineOffset(startLine - 1);
+      endLineStartOffset = endLine != startLine ? document.getLineOffset(endLine - 1) : startLineStartOffset;
+    } catch (BadLocationException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)